zhaquirks/tuya/tuya_trv.py

```diff
@@ -422,3 +474,200 @@ async def write_attributes(
+        type=t.int32s,
+        min_value=-6,
+        max_value=6,
+        unit=UnitOfTemperature.CELSIUS,
+        step=1,
+        translation_key="local_temperature_calibration",
+        fallback_name="Local temperature calibration",
+    )
+    .tuya_dp_multi(
+        dp_id=2,
+        attribute_mapping=[
+            DPToAttributeMapping(
+                ep_attribute=TuyaThermostatV2.ep_attribute,
+                attribute_name=TuyaThermostatV2.AttributeDefs.system_mode.name,
+                converter=lambda x: {
+                    TuyaPresetMode.Auto: Thermostat.SystemMode.Auto,
+                    TuyaPresetMode.Eco: Thermostat.SystemMode.Auto,
+                    TuyaPresetMode.Heat: Thermostat.SystemMode.Heat,
+                    TuyaPresetMode.Off: Thermostat.SystemMode.Off,
+                }[x],
+                dp_converter=lambda x: {
+                    Thermostat.SystemMode.Auto: TuyaPresetMode.Auto,
+                    Thermostat.SystemMode.Heat: TuyaPresetMode.Heat,
+                    Thermostat.SystemMode.Off: TuyaPresetMode.Off,
+                }[x],
+            ),
+            DPToAttributeMapping(
+                ep_attribute=TuyaMCUCluster.ep_attribute,
+                attribute_name="preset_mode",
+            ),
+        ],
+    )
+    .tuya_attribute(
+        dp_id=2,
+        attribute_name="preset_mode",
+        type=t.uint16_t,
+        is_manufacturer_specific=True,
+    )
+    .enum(
+        attribute_name="preset_mode",
+        cluster_id=TUYA_CLUSTER_ID,
+        enum_class=TuyaPresetMode,
+        translation_key="preset_mode",
+        fallback_name="Preset mode",
+    )
+    .tuya_battery(dp_id=6)
+    .tuya_switch(
+        dp_id=7,
+        attribute_name="child_lock",
+        translation_key="child_lock",
+        fallback_name="Child lock",
+    )
+    .tuya_number(
+        dp_id=9,
+        attribute_name="max_temperature",
+        type=t.uint16_t,
+        unit=UnitOfTemperature.CELSIUS,
+        min_value=15,
+        max_value=35,
+        step=1,
+        multiplier=0.1,
+        translation_key="max_temperature",
+        fallback_name="Max temperature",
+    )
+    .tuya_number(
+        dp_id=10,
+        attribute_name="min_temperature",
+        type=t.uint16_t,
+        unit=UnitOfTemperature.CELSIUS,
+        min_value=1,
+        max_value=15,
+        step=1,
+        multiplier=0.1,
+        translation_key="min_temperature",
+        fallback_name="Min temperature",
+    )
+    .tuya_switch(
+        dp_id=14,
+        attribute_name="window_detection",
+        translation_key="window_detection",
+        fallback_name="Open window detection",
+    )
+    .tuya_enum(
+        dp_id=15,
+        attribute_name="window_open",
+        enum_class=TuyaWindowOpen,
+        entity_platform=EntityPlatform.SENSOR,
+        entity_type=EntityType.STANDARD,
+        translation_key="window_open",
+        fallback_name="Window open",
+    )
```

Comment:
This seems like it should really be a binary sensor and not an analog/enum sensor.